src/util/selectMethod.ts

```diff
@@ -4,13 +4,18 @@ import { OptimizationOptions } from '../spectra-fitting';
 
 const LEVENBERG_MARQUARDT = 1;
 
+/** Algorithm to select the method.
+ * @param optimizationOptions - Optimization options
+ * @returns - The algorithm and optimization options
+ */
 export function selectMethod(optimizationOptions: OptimizationOptions = {}) {
   let { kind, options } = optimizationOptions;
   kind = getKind(kind);
   switch (kind) {
     case LEVENBERG_MARQUARDT:
       return {
         algorithm: levenbergMarquardt,
+        // check the options
```

Comment:
Please review the added comments because when a method is called `checkOptions` we should not add a comment telling it `check the options`.

If the function name requires a comment on the other hand maybe we should rename the function.